### DISCUSSION TITLE
동기와 비동기의 차이를 설명해주세요.

### DISCUSSION THREAD

**Discussion by eatingbug** [Team-Psyduck/tech-sharing#77] [General]
[1700번 멀티탭 스케줄링](https://www.acmicpc.net/problem/1700)

**Comment by SionBackEnd** [comment]
동기와 비동기는 주도권을 누가 갖고 있느냐로 구분할 수 있습니다. 
상황) A와 B 메서드가 존재하고, A메서드가 B메서드를 호출하는 상황을 가정하겠습니다. 

동기의 경우 A메서드가 B메서드를 호출 시 A가 갖고있던 주도권이 B에게 넘어가게 되어 B의 메서드가 종료되어 리턴되기 전까지 A는 주도권을 다시 얻지 못합니다. 
비동기의 경우 A메서드가 B메서드 호출 시 A가 갖고있던 주도권을 넘기는 것이 아니라, B를 호출하고 A의 메서드를 계속해서 실행합니다.
즉, A메서드는 B메서드가 종료될 때까지 기다리지 않고 계속해서 자신의 메서드를 진행할 수 있습니다.

**Comment by eatingbug** [comment]
- 동기는 한 작업이 완료될 때까지 다음작업을 대기하는 방식이다. 코드가 순차적으로 실행되며, 디버깅하기 쉽다는 장점이 있다.
- 비동기는 함수를 호출하거나 작업을 시작하고, 그 결과를 기다리지 않고 즉시 다음 작업으로 넘어가는 방식이다. 동기 방식과 다르게, 여러 작업을 병렬적으로 처리할 수 있다.

웹사이트에서 서버로 데이터를 요청하는 경우 동기방식으로 처리하면 서버로부터 데이터를 받아올 때까지 웹페이지가 대기상태에 머물게 된다. 그러나 비동기 방식으로 처리하게되면 서버로 데이터 요청을 보내고, 다른 작업을 수행하다가 응답을 처리할 수 있다.
